### PR TITLE
Disallow TAGSTRING_START after the conversion character in an f-string

### PR DESCRIPTION
**Pull request by lysnikolaou**
Opening this, just so you know what I meant in https://github.com/jimbaker/tagstr/issues/22#issuecomment-1524680437 and because I'll be on a plane the whole day tomorrow.

This PR solves 3/5 failures in `test_fstring` with the other two being related to what I wrote in https://github.com/jimbaker/tagstr/issues/22#issuecomment-1524693168.

### CODE REVIEW

**Comment by gvanrossum** [comment]
Nice, although this does make the code more complicated. It's already quite hard to follow what's going on -- I was mostly "programming by random modification" in the tag-strings-v2 branch.

**Comment by lysnikolaou** [comment]
That's true. Maybe it isn't worth it?

Either way, one of the thing on my to-do list is to start adding comments to the different parts of the tokenizer, where the logic may be hard to follow, like we discussed at PyCon.

**Review by M485631 — APPROVED** [review]
